### PR TITLE
Utilization for cronet operation results being propagated from Cronet native side

### PR DESCRIPTION
**Pull request by ssavant-pfpt**
- Added changes to utilize and propagate errors for any unsuccessful cronet operation on based of cronet results
- Added changes to check and avoid headers the way cronet checks on native side, and added helper logs for it

### CODE REVIEW

**Comment by ssavant-pfpt** [comment]
> * Added changes to utilize and propagate errors for any unsuccessful cronet operation on based of cronet results
> * Added changes to check and avoid headers the way cronet checks on native side, and added helper logs for it

@asmirnou 

1) Headers checking changes I have added for test only. I am debating now if we should keep it for future debugging use cases? 

2) Also thinking more about changes of utilizing cronet results, I feel they're not needed. 
Because our goal is to use enableCheckResults as true. So on native side as for any of the operation, if result is not success then native side is anyhow gonna crash even before returning to response to this method, let's say for **_urlRequest.InitWithParams_**
So essentially changes in PR only would be useful for debugging when enableCheckResults would be false. :) 

WDYT?

File: transport.go [hunk 1]
```diff
@@ -119,9 +127,16 @@ func (t *RoundTripper) RoundTrip(request *http.Request) (*http.Response, error)
...
 	responseHandler.request = urlRequest
-	urlRequest.InitWithParams(t.Engine, request.URL.String(), requestParams, callback, asyncExecutor)
+	cronetResult := urlRequest.InitWithParams(t.Engine, request.URL.String(), requestParams, callback, asyncExecutor)
+	if cronetResult != ResultSuccess {
+		return nil, errors.New("CRONET_ERROR : failed to init request")
+	}
 	requestParams.Destroy()
-	urlRequest.Start()
+	cronetResult = urlRequest.Start()
+	if cronetResult != ResultSuccess {
```
Comment (asmirnou) [inline transport.go:136]: the same check should be for `r.request.Read` below

**Review by asmirnou — COMMENTED** [review]
> WDYT?

1. Headers checking is needed for debugging the issue only.

2. Utilising Cronet results is needed at all times, because it helps to propagate the error code up to the calling party.  The presence of `enableCheckResults` will end at some point I think.

File: transport.go (hunk 1 above)
Comment (ssavant-pfpt) [reply transport.go:129]: Done

**Comment by ssavant-pfpt** [comment]
> > WDYT?
> 
> 1. Headers checking is needed for debugging the issue only.
> 2. Utilising Cronet results is needed at all times, because it helps to propagate the error code up to the calling party.  The presence of `enableCheckResults` will end at some point I think.

Added changes to tackle CronetResults for all 3 URL Request APIs.
And keeping http_util for now which can be used in future for debugging purpose. Thanks

File: transport.go [hunk 2]
```diff
@@ -119,9 +120,16 @@ func (t *RoundTripper) RoundTrip(request *http.Request) (*http.Response, error)
 	callback := NewURLRequestCallback(&responseHandler)
 	urlRequest := NewURLRequest()
 	responseHandler.request = urlRequest
-	urlRequest.InitWithParams(t.Engine, request.URL.String(), requestParams, callback, asyncExecutor)
+	cronetResult := urlRequest.InitWithParams(t.Engine, request.URL.String(), requestParams, callback, asyncExecutor)
+	if cronetResult != ResultSuccess {
+		return nil, errors.New("CRONET_ERROR : failed to init request")
```
[the remark this answers is not in the document]
Comment (ssavant-pfpt) [reply transport.go]: Done

File: transport.go [hunk 3]
```diff
@@ -119,9 +120,16 @@ func (t *RoundTripper) RoundTrip(request *http.Request) (*http.Response, error)
...
+	}
 	requestParams.Destroy()
-	urlRequest.Start()
+	cronetResult = urlRequest.Start()
+	if cronetResult != ResultSuccess {
+		urlRequest.Cancel()
```
[the remark this answers is not in the document]
Comment (ssavant-pfpt) [reply transport.go]: Done

File: transport.go [hunk 4]
```diff
@@ -17,8 +18,11 @@ import (
 
 var asyncExecutor Executor
 var syncExecutor Executor
+var logger *log.Logger
```
[the remark this answers is not in the document]
Comment (ssavant-pfpt) [reply transport.go]: Done

**Review by asmirnou — APPROVED** [review]

**Pull request merged**